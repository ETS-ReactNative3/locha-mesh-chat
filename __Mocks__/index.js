/* eslint-disable no-unused-vars */
/* eslint-disable max-classes-per-file */
/* eslint-disable import/no-extraneous-dependencies */
// eslint-disable-next-line no-unused-vars
import * as ReactNative from 'react-native';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mock from 'react-native-permissions/mock';
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';


// enzyme.configure({ adapter: new Adapter() });
jest.mock('react-native-fs', () => ({
  mkdir: jest.fn(),
  moveFile: jest.fn(),
  copyFile: jest.fn(),
  pathForBundle: jest.fn(),
  pathForGroup: jest.fn(),
  getFSInfo: jest.fn(),
  getAllExternalFilesDirs: jest.fn(),
  unlink: jest.fn(),
  exists: jest.fn(() => new Promise((resolve) => {
    resolve(true);
  })),
  stopDownload: jest.fn(),
  resumeDownload: jest.fn(),
  isResumable: jest.fn(),
  stopUpload: jest.fn(),
  completeHandlerIOS: jest.fn(),
  readDir: jest.fn(),
  readDirAssets: jest.fn(),
  existsAssets: jest.fn(),
  readdir: jest.fn(),
  setReadable: jest.fn(),
  stat: jest.fn(),
  readFile: jest.fn(),
  read: jest.fn(),
  readFileAssets: jest.fn(),
  hash: jest.fn(),
  copyFileAssets: jest.fn(),
  copyFileAssetsIOS: jest.fn(),
  copyAssetsVideoIOS: jest.fn(),
  writeFile: jest.fn(),
  appendFile: jest.fn(),
  write: jest.fn(),
  downloadFile: jest.fn(),
  uploadFiles: jest.fn(),
  touch: jest.fn(),
  MainBundlePath: jest.fn(),
  CachesDirectoryPath: jest.fn(),
  DocumentDirectoryPath: jest.fn(),
  ExternalDirectoryPath: jest.fn(),
  ExternalStorageDirectoryPath: jest.fn(),
  TemporaryDirectoryPath: jest.fn(),
  LibraryDirectoryPath: jest.fn(),
  PicturesDirectoryPath: jest.fn(),
}));


jest.mock('react-native-share', () => {
  // code here
});

jest.mock('react-native-sound', () => class Sound {
  constructor(path, enconde, callback) {
    this.path = path;
    this.enconde = enconde;
    callback(false);
  }

  getDuration = () => 20

  pause = () => {
    jest.fn();
  }
});


jest.mock('react-native-gesture-handler', () => {
  // code here
});

jest.mock('react-navigation-stack', () => {
  // code here
});


jest.mock('react-native-audio', () => {
  // code here
});

jest.mock('react-native-permissions', () => mock);

jest.mock('@react-native-community/slider', () => {
  const data = () => null;
  return data;
});


jest.mock('react-navigation', () => ({
  createAppContainer: jest.fn().mockReturnValue(() => null),
  createDrawerNavigator: jest.fn(),
  createMaterialTopTabNavigator: jest.fn(),
  createStackNavigator: jest.fn(),
  StackActions: {
    push: jest.fn().mockImplementation((x) => ({ ...x, type: 'Navigation/PUSH' })),
    replace: jest.fn().mockImplementation((x) => ({ ...x, type: 'Navigation/REPLACE' })),
  },
  NavigationActions: {
    navigate: jest.fn().mockImplementation((x) => x),
  }
}));

jest.mock('react-native-background-timer', () => {
  // code here
});

jest.mock('rn-fetch-blob', () => ({
  config: jest.fn().mockReturnValue(() => {
    jest.fn().mockReturnValue(() => new Promise());
  })
}));


jest.mock(
  '../node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter',
);


jest.mock('native-base/dist/src/basic/Icon', () => jest.genMockFromModule('native-base/dist/src/basic/Icon'));


jest.mock('react-native/Libraries/Utilities/Platform', () => ({
  OS: 'ios',
  select: jest.fn((selector) => selector.ios),
}));

jest.mock('@react-native-community/async-storage', () => ({
  getItem: jest.fn((data) => new Promise((resolve) => {
    if (data === 'upnp') {
      resolve(true);
    } else {
      resolve(null);
    }
  })),

  setItem: jest.fn(() => new Promise((resolve) => {
    resolve(true);
  })),
  removeItem: jest.fn(() => new Promise((resolve) => {
    resolve();
  }))
}));

jest.doMock('react-native', () =>
  // Extend ReactNative
  Object.setPrototypeOf(
    {
      // Mock a native module
      NativeModules: {
        ...ReactNative.NativeModules,
        Override: { great: 'success' },
        LocalNotification: {
          requestPermission: jest.fn(),


        },
        RNDeviceInfo: {
          VersionInfo: '1',
          getIpv6Andipv4Adress: jest.fn().mockReturnValue(['192.168.0.1'])
        },
        ChatService: {
          start: jest.fn(() => new Promise((resolve) => {
            resolve('5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1');
          })),
          stop: jest.fn(() => new Promise((resolve) => { resolve(); })),
          addNewAddressListen: jest.fn(() => new Promise((resolve) => { resolve(); }))
        },
        bitcoinModule: {
          createWallet: jest.fn(() => new Promise((resolve) => {
            resolve('5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1');
          })),
          sha256: new Promise((resolve) => { resolve('5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1'); }),
          getPrivateKey: jest.fn(() => new Promise((resolve) => { resolve('test'); }))
        },
        RNPermissions: {
          ...mock
        }
      },
      StyleSheet: {
        create: () => ({}),
        flatten: () => ({})
      },
      Platform: {
        OS: 'ios',
        select: jest.fn((selector) => selector.ios),
      }
    },
    ReactNative,
  ));
