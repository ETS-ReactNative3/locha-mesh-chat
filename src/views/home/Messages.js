import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableNativeFeedback,
  TouchableHighlight,
  Image,
  Platform
} from 'react-native';
import Moment from 'moment';
import { Thumbnail, Icon } from 'native-base';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { getIcon, hashGenerateColort } from '../../utils/utils';
import Player from '../../components/Player ';

const TouchableList = Platform.select({
  ios: () => TouchableHighlight,
  android: () => TouchableNativeFeedback,
})();

export const ReceiveMessage = ({
  onClick,
  onSelected,
  userInfo,
  item,
  contactInfo,
  selected
}) => (
  <TouchableList
    onLongPress={() => onSelected(item)}
    onPress={() => onClick(item)}
    style={{
      marginVertical: 5,
      minHeight: 70,
      width: '100%',
      flexDirection: 'row'
    }}
  >
    <View style={[styles.receiveContainer, selected]}>
      {item.toUID === 'broadcast' && !contactInfo && (
      <Thumbnail
        style={{
          marginLeft: 5,
          marginTop: 5
        }}
        source={{
          uri: `${getIcon(item.fromUID)}`
        }}
      />
      )}

      {item.toUID === 'broadcast' && contactInfo && (
      <Thumbnail
        style={{
          marginLeft: 5,
          marginTop: 5
        }}
        source={{
          uri: `${userInfo.picture ? userInfo.picture : getIcon(item.fromUID)
          }`
        }}
      />
      )}

      <View style={{ width: '90%', flexDirection: 'row' }}>
        <View style={styles.textContent1}>
          {item.name && (
          <Text
            style={{
              paddingBottom: 7,
              color: hashGenerateColort(item.fromUID)
            }}
          >
            {userInfo.name}
          </Text>
          )}
          <View style={{ minWidth: 110 }}>
            {item.file && (
            <View style={{ minWidth: '80%' }}>
              <Image
                style={{ width: '100%', height: 150 }}
                source={{
                  resizeMode: 'contain',
                  uri: item.file.file,
                  cache: 'force-cache'
                }}
              />
            </View>
            )}
            <Text style={{ fontSize: 15 }}>{item.msg}</Text>
          </View>
          <Text
            style={{
              paddingTop: 3,
              paddingLeft: 10,
              paddingBottom: 6,
              fontSize: 12,
              textAlign: 'right'
            }}
          >
            {Moment(Number(item.timestamp)).format('LT')}
          </Text>
        </View>
      </View>
    </View>
  </TouchableList>
);

export const SenderMessage = ({
  onClick,
  onSelected,
  item,
  selected,
  retry
}) => {
  const timeCreated = Moment(item.shippingTime);
  const cancelled = !!((Moment().diff(timeCreated, 's') > 30 && item.status === 'pending')
    || item.status === 'not sent');

  const styleBody = item.msg.length < 20 ? styles.styleBody1 : styles.styleBody2;

  const textStyle = item.msg.length < 20 ? styles.textStyle1 : styles.textStyle2;

  const iconName = item.toUID ? 'checkmark' : 'user-check';
  const IconType = iconName === 'checkmark' ? 'Ionicons' : 'FontAwesome5';
  return (
    <TouchableList
      underlayColor="#DDD"
      style={{
        marginVertical: 5,
        width: '100%',
        justifyContent: 'flex-end',
        flexDirection: 'row'
      }}
      onLongPress={() => onSelected(item)}
      onPress={() => onClick(item)}
    >
      <View style={[styles.senderContainer, selected]}>
        <View style={styles.textContent2}>
          {item.file && (
            <View style={{ minWidth: '80%' }}>
              <Image
                style={{ width: '100%', height: 150 }}
                source={{
                  resizeMode: 'contain',
                  uri: item.file.file,
                  cache: 'force-cache'
                }}
              />
            </View>
          )}
          <View style={styleBody}>
            <View>
              <Text style={{ fontSize: 15 }}>{item.msg}</Text>
            </View>
            <View style={textStyle}>
              <View style={{ flexDirection: 'row' }}>
                <Text>{Moment(Number(item.timestamp)).format('LT')}</Text>
                {item.status === 'pending' && !cancelled && (
                  <Icon
                    style={{ color: 'gray', fontSize: 15, marginLeft: 10 }}
                    name="time"
                  />
                )}
                {item.status === 'delivered' && (
                  <Icon
                    style={{ color: 'gray', fontSize: 15, marginLeft: 10 }}
                    type={IconType}
                    name={iconName}
                  />
                )}
                {item.status === 'read' && (
                  <Icon
                    style={{ color: 'gray', fontSize: 15, marginLeft: 10 }}
                    name="done-all"
                  />
                )}
                {cancelled && (
                  <TouchableOpacity onPress={() => retry(item)}>
                    <Icon
                      type="MaterialIcons"
                      style={{ color: 'gray', fontSize: 16, marginLeft: 10 }}
                      name="error"
                    />
                  </TouchableOpacity>
                )}
              </View>
            </View>
          </View>
        </View>
      </View>
    </TouchableList>
  );
};

export const SoundMessage = ({
  onClick,
  onSelected,
  userInfo,
  item,
  contactInfo,
  selected,
  rule,
  index,
  chats
  // eslint-disable-next-line consistent-return
}) => {
  if (!rule) {
    return (
      <TouchableList
        key={index}
        underlayColor="#DDD"
        onLongPress={() => onSelected(item)}
        onPress={() => onClick(item)}
        style={{
          marginVertical: 5,
          minHeight: 70,
          width: '100%',
          flexDirection: 'row'
        }}
      >
        <View style={[styles.receiveContainer, selected]}>
          {!item.toUID && !contactInfo && (
            <Thumbnail
              style={{
                marginLeft: 5,
                marginTop: 5
              }}
              source={{
                uri: `${getIcon(item.fromUID)}`
              }}
            />
          )}

          {!item.toUID && contactInfo && (
            <Thumbnail
              style={{
                marginLeft: 5,
                marginTop: 5
              }}
              source={{
                uri: userInfo.picture
                  ? `${userInfo.picture}`
                  : `${getIcon(item.fromUID)}`
              }}
            />
          )}
          <View style={{ width: '90%', flexDirection: 'row' }}>
            <View style={styles.textContent1}>
              {item.name && (
                <Text
                  style={{
                    paddingBottom: 7,
                    color: hashGenerateColort(item.fromUID)
                  }}
                >
                  {userInfo.name}
                </Text>
              )}
              <Player path={item.file.file} />
              <Text
                style={{
                  paddingTop: 3,
                  paddingLeft: 10,
                  paddingBottom: 6,
                  fontSize: 12,
                  textAlign: 'right'
                }}
              >
                {Moment(Number(item.timestamp)).format('LT')}
              </Text>
            </View>
          </View>
        </View>
      </TouchableList>
    );
  }
  if (item.file.file === chats[index].file.file) {
    return (
      <TouchableList
        key={index}
        underlayColor="#DDD"
        useForeground
        style={{
          marginVertical: 5,
          width: '100%',
          justifyContent: 'flex-end',
          flexDirection: 'row'
        }}
        onLongPress={() => onSelected(item)}
        onPress={() => onClick(item)}
      >
        <View style={[styles.senderContainer, selected]}>
          <View style={styles.textContent2}>
            <Player path={item.file.file} index={index} />
            <Text
              style={{
                paddingTop: 7,
                paddingLeft: 5,
                paddingBottom: 6,
                fontSize: 12,
                textAlign: 'right'
              }}
            >
              {Moment(Number(item.timestamp)).format('LT')}
            </Text>
          </View>
        </View>
      </TouchableList>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    minHeight: '100%',
    backgroundColor: '#eeeeee',
    paddingBottom: 10
  },
  senderContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    paddingBottom: 10
  },

  selected: {
    backgroundColor: 'rgba(255, 235, 59 , 0.5)'
  },

  receiveContainer: {
    width: '100%',
    alignItems: 'flex-start',
    marginBottom: 10,
    flexDirection: 'row'
  },
  textContent1: {
    maxWidth: '82%',
    backgroundColor: '#fff',
    minHeight: 30,
    paddingTop: 5,
    paddingHorizontal: 10,
    marginHorizontal: 10,
    borderRadius: 7,
    flexDirection: 'column'
  },
  textContent2: {
    maxWidth: '82%',
    backgroundColor: '#fff',
    minHeight: 30,
    paddingTop: 5,
    paddingHorizontal: 10,
    marginHorizontal: 10,
    borderRadius: 7,
    flexDirection: 'column',
  },
  styleBody1: {
    flexDirection: 'row',
    flex: 1
  },
  styleBody2: {
    flexDirection: 'column'
  },

  textStyle1: {
    paddingTop: 7,
    paddingLeft: 10,
    paddingBottom: 6,
    fontSize: 12,
    alignItems: 'flex-end'
  },

  textStyle2: {
    paddingTop: 5,
    paddingBottom: 6,
    flex: 1,
    alignItems: 'flex-end'
  }
});
