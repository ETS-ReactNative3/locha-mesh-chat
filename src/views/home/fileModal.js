import React, { Component } from 'react';
import Modal from 'react-native-modal';
import {
  View, Text, TouchableOpacity, Dimensions
} from 'react-native';
import { Thumbnail } from 'native-base';
import ImagePicker from 'react-native-image-crop-picker';
import RNFS from 'react-native-fs';
import { images } from '../../utils/constans';
import { FileDirectory } from '../../utils/utils';

/**
 *
 *
 * @export
 * @class EditPhoto
 * @description modal to select the type of file to send
 * @extends {Component}
 */

export default class FileModal extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  /**
   *
   * Search the photos in the gallery and send it to the chat image viewer
   * @memberof FileModal
   */

  getPhotosFromGallery = () => {
    const imageArray = [];
    // this.props.close()
    ImagePicker.openPicker({
      multiple: false,
      includeBase64: true,
      width: 200,
      height: 200
    }).then((image) => {
      image.forEach((data) => {
        imageArray.push({
          url: data.path,
          base64: data.data,
          width: Dimensions.get('window').width
        });
      });
      this.props.setImageView(imageArray);
      this.props.close();
    });
  };

  /**
   *
   * take the photo, cut it, move the application folder and send it in chat image viewer
   * @memberof FileModal
   */

  GetphotoFromCamera = () => {
    this.props.close();
    ImagePicker.openCamera({
      width: 200,
      height: 200,
      cropping: false,
      includeBase64: true
    }).then((image) => {
      const newPath = `${FileDirectory}/Pictures/IMG_${new Date().getTime()}.jpg `;
      // RNFS.moveFile(image.path, newPath).then(() => {
      setTimeout(() => {
        const imagesView = [
          {
            url: image.path,
            base64: image.data,
            width: Dimensions.get('window').width
          }
        ];
        this.props.setImageView(imagesView);
        this.props.close();
      }, 200);
      // });
    });
  };


  render() {
    const { open, screenProps } = this.props;
    return (
      <View>
        <Modal
          style={{
            justifyContent: 'flex-end',
            margin: 10,
            marginBottom: 54
          }}
          isVisible={open}
          onBackdropPress={() => this.props.close()}
          swipeDirection={['up', 'left', 'right', 'down']}
          backdropOpacity={0}
        >
          <View
            style={{
              backgroundColor: '#fff',
              width: '100%',
              height: '20%',
              alignItems: 'center',
              borderRadius: 5,
              flexDirection: 'row',
              justifyContent: 'space-evenly'
            }}
          >
            <TouchableOpacity
              style={{ alignItems: 'center' }}
              onPress={() => {
                this.getPhotosFromGallery();
              }}
            >
              <Thumbnail source={images.file.url} />
              <Text>{screenProps.t('Settings:gallery')}</Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={{ alignItems: 'center' }}
              onPress={() => {
                this.GetphotoFromCamera();
              }}
            >
              <Thumbnail source={images.camera.url} />
              <Text>{screenProps.t('Settings:camera')}</Text>
            </TouchableOpacity>
          </View>
        </Modal>
      </View>
    );
  }
}
