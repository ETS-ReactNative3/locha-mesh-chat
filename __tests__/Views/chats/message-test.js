import '../../../__Mocks__';
import React from 'react';
import { shallow } from 'enzyme';
import { ReceiveMessage } from '../../../src/views/home/Messages';

const mockMessage = [{
  fromUID: '16Uiu2HAm9JHK1iG6g2nLNw2fmAWerGiVoKMnjcvEs6pJVnd6STPC',
  id: 'ad7ae2f9dc672c55095e60f2e8367393261088972af35c49c007be0c642fda24',
  msg: 'Hola',
  msgID: 'ad7ae2f9dc672c55095e60f2e8367393261088972af35c49c007be0c642fda24',
  name: undefined,
  shippingTime: 1602013395803,
  status: 'pending',
  time: 1602013395803,
  timestamp: 1602013395786,
  toUID: 'broadcast',
  type: 1
}];

const screenProps = {
  t: (data) => data
};

const mockContact = [{
  hashUID: '1ed4091506f77825bccb4dadcf89f1b62ff3257d2afc3886bf859c0330120a27',
  name: 'Luis',
  nodeAddress: '/ip4/85.53.137.144/tcp/4444',
  picture: null,
  uid: '16Uiu2HAm2BXA1TTUE7vvSu3U4YURxb4FwTYjfrwnZN618jrYBpNS'
}];

const onLongPressMock = jest.fn();
const onPressMock = jest.fn();

describe('message component', () => {
  const wrapper = shallow(
    <ReceiveMessage
      screenProps={screenProps}
      item={mockMessage}
      contactInfo={mockContact}
      onClick={onPressMock}
      onSelected={onLongPressMock}

    />
  );

  test('render all component', () => {
    expect(wrapper.instance()).toBeDefined();
  });

  test('onLongPress button message', () => {
    wrapper.find('ForwardRef').at(0).props().onLongPress();
    expect(onLongPressMock.mock.calls.length).toBe(1);
  });

  test('onPress button message', () => {
    wrapper.find('ForwardRef').at(0).props().onPress();
    expect(onPressMock.mock.calls.length).toBe(1);
  });
});
