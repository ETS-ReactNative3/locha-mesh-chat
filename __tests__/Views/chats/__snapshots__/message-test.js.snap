// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message component render SenderMessage 1`] = `
<ForwardRef
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginVertical": 5,
      "width": "100%",
    }
  }
  underlayColor="#DDD"
>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <View>
        <View>
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Hola
          </Text>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text>
              3:43 PM
            </Text>
            <ForwardRef
              onPress={[Function]}
            >
              <Styled(Icon)
                name="error"
                style={
                  Object {
                    "color": "gray",
                    "fontSize": 16,
                    "marginLeft": 10,
                    "marginTop": 2,
                  }
                }
                type="MaterialIcons"
              />
            </ForwardRef>
          </View>
        </View>
      </View>
    </View>
  </View>
</ForwardRef>
`;
